### ISSUE TITLE
Error opening shp file

### ISSUE THREAD

**Issue by amaschke** [Toblerity/Fiona#525]
This is my first time opening an issue, please don't yell at me. Now that that's out of the way, I was researching an error I was receiving when trying to read a shapefile in using Geopandas in a Jupyter Python 3 notebook. While researching the error, I came across issue #457 and I updated my fiona version to 1.7.11 as advised in the issue referenced. I reran all the code in my notebook, but was given the same error again. Please advise whether this is a returned bug or I'm completely misunderstanding the error, if so, any guidance towards a solution would be highly appreciated.

## Expected behavior and actual behavior:

I expected to read in a shapefile and be presented with a table. 

Instead, I received the following error:

```
TypeError                                 Traceback (most recent call last)
<ipython-input-10-41e8efd48b13> in <module>()
----> 1 blocks = gpd.read_file("gz_2010_06_150_00_500k.shp")
      2 blocks.head()

/usr/local/lib/python3.6/site-packages/geopandas/io/file.py in read_file(filename, **kwargs)
     19     """
     20     bbox = kwargs.pop('bbox', None)
---> 21     with fiona.open(filename, **kwargs) as f:
     22         crs = f.crs
     23         if bbox is not None:

/usr/local/lib/python3.6/site-packages/fiona/__init__.py in open(path, mode, driver, schema, crs, encoding, layer, vfs, enabled_drivers, crs_wkt)
    163         c = Collection(path, mode, driver=driver, encoding=encoding,
    164                        layer=layer, vsi=vsi, archive=archive,
--> 165                        enabled_drivers=enabled_drivers)
    166     elif mode == 'w':
    167         if schema:

/usr/local/lib/python3.6/site-packages/fiona/collection.py in __init__(self, path, mode, driver, schema, crs, encoding, layer, vsi, archive, enabled_drivers, crs_wkt, **kwargs)
    151             if self.mode == 'r':
    152                 self.session = Session()
--> 153                 self.session.start(self)
    154             elif self.mode in ('a', 'w'):
    155                 self.session = WritingSession()

fiona/ogrext.pyx in fiona.ogrext.Session.start (fiona/ogrext2.c:8430)()

fiona/_err.pyx in fiona._err.GDALErrCtxManager.__exit__ (fiona/_err.c:1809)()

fiona/_err.pyx in fiona._err.CPLE_BaseError.__init__ (fiona/_err.c:1242)()

TypeError: __init__() takes exactly 4 positional arguments (2 given)

## Here's my code: 

states = gpd.read_file("cb_2016_us_state_500k/cb_2016_us_state_500k.shp")
states.head()
```

## Operating system

Mac OS X 10.12

## Fiona version

1.7.11

**Comment by sgillies** [comment]
@amaschke will you please add `import fiona; print(fiona.__version__)` to the top of your script and run it again? I think you might not be using Fiona 1.7.11 (which should not raise this error).

**Comment by amaschke** [comment]
Thank you for your response. I did update it, but it seems like part of the error was on my end, as I made the rookie error of removing the shp file out of its group of files. I thought I was still getting the same error after updating fiona, but it may have just been the new error because I removed the shp file from other components (still learning a lot over here!). Long story short, it has been resolved, thank you so much for responding.
